### ISSUE TITLE
Add completion for true/false 

### ISSUE THREAD

**Issue by neonaot** [intellij-rust/intellij-rust#8471]
## Environment

* **IntelliJ Rust plugin version:** 0.4.165.4395-213-nightly

## Problem description
Completion does not work for variables without a type specification
<img width="593" alt="image" src="https://user-images.githubusercontent.com/49211026/151162945-0d594d18-587c-432a-a474-f41593769166.png">
<img width="421" alt="image" src="https://user-images.githubusercontent.com/49211026/151163958-e5e1faf3-4b01-4546-987a-98cb03db009e.png">

Python and Kotlin have this completion: 
<img width="593" alt="image" src="https://user-images.githubusercontent.com/49211026/151163016-13e66c33-b1cf-450b-9d1f-4ca7d91c0183.png">

**Comment by afetisov** [comment]
I'd say it's by design. Completing true/false for every untyped variable would either give lots of annoying false positives, or end up with those items low in the completion list, making them hard to find anyway. Kotlin and Python don't have that problem, because the spelling True/False starts with a capital letter, which is by convention not the way to name functions and variables. This means that uppercase-T is extremely likely to be the start of True.

**Comment by Undin** [comment]
> Kotlin and Python don't have that problem, because the spelling True/False starts with a capital letter

Just want to note that Kotlin uses the same form of `true` and `false` values as Rust

**Comment by afetisov** [comment]
Oh, right, indeed. How does the Kotlin plugin deal with false positives?

**Comment by Undin** [comment]
> How does the Kotlin plugin deal with false positives?

I don't think that they do anything with it. And I have never noticed that it somehow annoys me

**Comment by neonaot** [comment]
For me, it is a little bit annoying when I type something, but completion does not suggest the expected variant, and it makes me think that I'm writing something incorrectly.  
typing `tr` -> no completion -> "oh no, maybe I forgot the right case again" ->  retyping `Tr` -> no completion -> panic!

**Comment by afetisov** [comment]
But it's even more annoying when the completion substitutes the wrong thing. For example, current completion likes to insert the perma-unstable `panic()` function instead of `panic!()` macro, or `.ok` postfix template instead of `.ok()` function on `Option`. At least with those completions there are clear rules which could be used to rank them lower (a stable macro should rank higher than an unstable function, an inherent function should have higher priority than a generic template). With `true/false` I don't see any reasonable rules to choose them vs some other function with similar initials (note that `tr` can also mean `to_response`), other than cross fingers and hope that ML completion does magic.
